### ISSUE TITLE
[Research] Evaluate TVDB API v2

### ISSUE THREAD

**Issue by lokenx** [lokenx/plexrequests-meteor#149]
Refer to ticket #132

**Comment by camjac251** [comment]
Have you made any progress on this? Just today I searched for a show on themoviedb and thetvdb. TVdb found it but themoviedb couldnt.

**Comment by lokenx** [comment]
I've been having a cursory look but haven't had time to get a working demo going yet. You still have to be careful though, as the website and API search for TheTVDB are separate and may still provide different results.

**Comment by lokenx** [comment]
I'm happy with the search results of the TVDB new API so shall be incorporating it once it's fully live. Not sure if that'll be in this meteor version or in the next node/python version but we'll see.

**Comment by Dahlgren** [comment]
I tried yesterday to add check if movie/series already exists while searching and this would solve the issue with fetching the TVDb id since it's not returned when searching with TMDb. Loading the id for each item takes to long to be feasible.

**Comment by lokenx** [comment]
Yeah each would require a separate call. I will be switching to the TVDB api as it seems much better and is providing proper results now.

It's still not live though and tokens expire after 1hr/24 hours so once that pushes/is live I can work on it. Although at this point I'll likely be adding it to my fork/rewrite first.

**Comment by lokenx** [comment]
Unrelated, I'm also going to switch to Plex to check for availability (not on search though, that will still remain with the flow as is),

**Comment by Dahlgren** [comment]
Sounds great!

**Comment by rigrassm** [comment]
It looks like TVDB's API keys are now persistent (at least from what I read it seems that way). 

I'd be willing to start working on a meteor module to integrate this to improve search results for TV series'. I had a case yesterday where I was able to find a series in the PlexRequests channel (which uses TVDB) after Plex requests failed to. Though I plan on having both available, it'd be nice to have consistent search results regardless of which client users decide to use.

I've taken a cursory look at the api and the only major difference is TVDB returns an XML response which will mean adding another package to convert that to js. 

Let me know if you think it'll be worth it to get this going and I'll jump on it today.

**Comment by lokenx** [comment]
Oh OK that's good to know! I actually already got it working with my pure node version of plex-requests [here](https://github.com/lokenx/plex-requests/tree/master/lib/plugins/search). It does json if you tell it to.

I stalled working on it due to time constraints. But if you form the current branch, and use the code linked above to add in the TVDB for searching can definitely do that. Other option is to finish up the node version (it just doesn't tie into download providers or have a front end).

My biggest problem is purely time constraints right now :disappointed:
